src/build.py

```diff
@@ -1218,18 +1206,41 @@ def TestEmwasm():
       outdir=emscripten_wasm_out)
 
 
+def ExecuteEmscriptenTestSuite(name, config, outdir, warn_only):
+  buildbot.Step('Execute emscripten testsuite (%s)' % name)
+  Mkdir(outdir)
+  try:
+    proc.check_call(
+        [sys.executable,
+         os.path.join(INSTALL_DIR, 'emscripten', 'tests', 'runner.py'),
+         'binaryen2', '--em-config', config],
+        cwd=outdir)
+  except proc.CalledProcessError:
+    buildbot.Fail(warn_only)
+
+
 def TestEmtest():
   ExecuteEmscriptenTestSuite(
       'emtest',
       EMSCRIPTEN_CONFIG_WASM,
-      EMSCRIPTEN_TEST_OUT_DIR)
+      EMSCRIPTEN_TEST_OUT_DIR,
+      warn_only=True)
+
+
+def TestEmtestAsm2Wasm():
+  ExecuteEmscriptenTestSuite(
+      'emtest-asm2wasm',
+      EMSCRIPTEN_CONFIG_ASMJS,
+      EMSCRIPTEN_ASMJS_TEST_OUT_DIR,
+      warn_only=False)
 
 
 ALL_TESTS = [
     Test('bare', TestBare),
     Test('asm', TestAsm),
     Test('emwasm', TestEmwasm),
     Test('emtest', TestEmtest),
+    Test('emtest-asm2wasm', TestEmtestAsm2Wasm),
```

Comment:
The above means that "emtest" is only used as an abbreviation for the command line, which I think was the original intent? Now here we sort of have a sparse 2D matrix with test suite (torture, emscripten) x codegen (bare, asm2wasm, emwasm). `bare` is torture+bare, `asm` is torture+asm2wasm, `emwasm` is torture+emwasm, `emtest` is emscripten+emwasm, and now the new emscripten+asm2wasm. 

The intent was to keep these names short since we (at least I) type them on the command line semi-regularly; I don't know if that's more important than organization here or what. Just musing I guess, for now maybe this could be called `emtest-asm` to match `asm` above.